iquip/apps/scheduler.py

```diff
@@ -219,10 +217,64 @@ def paint(self,
     def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:  # pylint: disable=unused-argument
         """Overridden."""
         data = index.data(Qt.DisplayRole)
+        if data is None:
+            return QSize(0, 0)
         experimentView = ExperimentView(data)
         return experimentView.sizeHint()
 
 
+class _ExperimentQueueFetcherThread(QThread):
+    """QThread for fetching the queued experiments from the proxy server.
+
+    Signals:
+        fetched(experimentList, runningExperiment):
+          The experiment queue and currently running experiment are fetched.
+    """
+
+    fetched = pyqtSignal(list, object)
+
+    def __init__(
+        self,
+        callback: Callable[[SubmittedExperimentInfo, None], List[SubmittedExperimentInfo]],
+        parent: Optional[QObject] = None
+    ):
+        """Extended.
+
+        Args:
+            callback: The callback method called after this thread is finished.
+        """
+        super().__init__(parent=parent)
+        self.fetched.connect(callback, type=Qt.QueuedConnection)
+
+    def run(self):
+        """Overridden.
+
+        Fetches the experiment list as a dictionary from the proxy server,
+        and emits a list and an ExperimentInfo instance for display.
+        """
+        while True:
+            try:
+                response = requests.get("http://127.0.0.1:8000/experiment/queue/", timeout=10)
+                response.raise_for_status()
+                response = response.json()
+            except requests.exceptions.Timeout:
+                continue
+            except requests.exceptions.RequestException as err:
+                print(err)
+                return
+            runningExperiment = None
+            experimentList = []
+            for key, value in response.items():
+                experimentInfo = SubmittedExperimentInfo(rid=int(key))
+                for item in tuple(item for item, _ in experimentInfo.items()):
+                    setattr(experimentInfo, item, value[item])
+                if value["status"] in ["running", "run_done", "analyzing", "deleting"]:
```

Comment:
How can you find these four statuses? Is there any reference?

Reply:
Also, I think one experiment may be being analyzed while the other is running.

Reply:
you can find the statuses here: [`artiq/artiq/master/scheduler.py`](https://github.com/m-labs/artiq/blob/01d777c977083c9166570eb26bc43112cecfde0a/artiq/master/scheduler.py#L18)

Reply:
Then I will change it to detect only two states, `running`, `run_done`. I can add them later in another list view mentioned in #106.